src/richtext/utils/serializerHelpers.ts

```diff
@@ -0,0 +1,158 @@
+import { Element } from "hast";
+import { toHtml } from "hast-util-to-html";
+
+import { OEmbedType } from "../../types/value/embed";
+import { LinkType } from "../../types/value/link";
+import {
+	RTEmbedNode,
+	RTImageNode,
+	RTLinkNode,
+} from "../../types/value/richText";
+
+import { RTPartialInlineNode } from "./RichTextFieldBuilder";
+
+/**
+ * A warning thrown when a serializer encounters an issue while working with an
+ * hast {@link Element} node. It's then added to the vfile messages.
+ */
+export class SerializerWarning extends Error {}
+
+/**
+ * Serializes an hast {@link Element} node to a {@link RTImageNode}.
+ *
+ * @param node - An hast node to serialize
+ *
+ * @returns Equivalent {@link RTImageNode}
+ */
+export const serializeImage = (node: Element): RTImageNode => {
+	const src = node.properties?.src as string | undefined;
+
+	if (!src) {
+		throw new SerializerWarning(
+			"Element of type `img` is missing an `src` attribute",
+		);
+	}
+
+	const url = new URL(src);
+
+	let width = (node.properties?.width as number) ?? 0;
+	let height = (node.properties?.height as number) ?? 0;
```

Comment:
I updated them to `undefined` but I had to cast them as `number` down the road because the API types are always filled. Does that sound OK to you? Should we update the API types instead to allow `undefined`?

Reply:
Hm… the public API always returns dimension values so we should keep them as always defined.

How do we know the Migration API doesn't require them? If the API really doesn't require them, then casting is good with me. When we start working on the Migration API client, we may want to look into a custom Migration API-specific `ImageField` type.

Reply:
Updated to cast! That's what @Heyrows shared with me for reference: https://github.com/prismicio/migration-api/blob/5a8455118c5fcbcb60225a94919a97cc7646fd73/src/models/import/validators/fields/nestable/Image/merge.ts#L35-L38, they leave them empty and compute them later on.

I don't have an issue with having a separate `ImageField` type, but then that means we also need to make the rich text types and nodes more flexible to accommodate both or duplicate them, which I feel meh about 🤔

Reply:
Added a small comment about it linking to this thread 👌